### PR TITLE
Improving MathML transpiling

### LINKED ISSUE #51: Rewrite transpiler to use etree instead of dom
Investigate why the mathml2sympy transpiler doesn't like a reference to the MathML DOM element. Currently passing string representation of the MathML.

See `_add_maths` in `parser.Parser`

### PR DESCRIPTION
**Pull request by jonc125**
## Description
Moved the transpiler into parser.py and made it use lxml (elementtree API) rather than DOM. This means we don't need to serialise MathML to string & re-parse to convert to sympy.

## Motivation and Context
Fixes #51. Should also help with #208.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue) - at least, it's non-breaking provided users haven't accessed transpiler.py directly!

## Checklist:
- [x] I have updated all documentation necessary.
- [x] I have checked spelling in (new) comments.

## Testing
- [x] Testing is done automatically and codecov shows test coverage

### CODE REVIEW

File: tests/test_model_evaluation.py [hunk 1]
```diff
@@ -1,90 +1,76 @@
...
-
 import networkx
 import pytest
 import sympy
 
-from cellmlmanip import load_model
-
 
-class TestModelEvaluation():
+def test_model_evaluation(hh_model):
```
Comment (MichaelClerx) [inline tests/test_model_evaluation.py:6]: This could probably go in somewhere else then? Or maybe leave for a bigger test reorganisation :-)

**Comment by MichaelClerx** [comment]
> Also (but maybe less pressing), I found that if you use an assert instead of raising a value error it plays nicely with the code coverage, so e.g.
> `assert len(result) == 2, 'Need exactly 2 children for <piece>'`
> which f triggered gives:
> `AssertionError: Need exactly 2 children for <piece>`
> as opposed to

Asserts should never go from cellmlmanip to a cellmlmanip user though. They're sanity checks, not part of the the public API.

**Comment by MichaelClerx** [comment]
Also, coverage and style checks are there to help. So adapting the code for no other reason than to make them pass is a terrible idea :D

**Comment by MichaelClerx** [comment]
I even try and write commit messages like "improved code style" rather than "fixed flake8", just to remind myself that flake8 is a tool, not the goal :-|
(sorry for being so boring)

**Comment by MauriceHendrix** [comment]
> Also, coverage and style checks are there to help. So adapting the code for no other reason than to make them pass is a terrible idea :D

Sure, but really in my opinion an `assert` is a better fit for something that should be true but where you think it feels like something that might go wrong in the future, even if there is no way of making it fail right now. 
I would say `raise` is more appropriate for errors you are actively expecting and handling.

**Comment by MichaelClerx** [comment]
This one's asking how many children in a `<piece>` though, that's determined by the CellML file you give it as input, isn't it? Might be that this gets caught by an earlier validation, which would make it unreachable and in that case I fully agree with you :D

**Comment by MauriceHendrix** [comment]
> This one's asking how many children in a `<piece>` though, that's determined by the CellML file you give it as input, isn't it? Might be that this gets caught by an earlier validation, which would make it unreachable and in that case I fully agree with you :D

Well if it doesn't then it would need a test contrived cellml file as case no?

**Pull request merged**